brainiak/fcma/util.py

```diff
@@ -0,0 +1,95 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Full Correlation Matrix Analysis (FCMA)
+
+Correlation related high performance routines
+"""
+
+# Authors: Yida Wang
+# (Intel Labs), 2017
+
+import numpy as np
+from . import cython_blas as blas
+from scipy.stats.mstats import zscore
+import math
+
+
+def normalize_for_correlation(data, axis):
+    """normalize the data before computing correlation
+
+    The data will be z-scored and divided by sqrt(n)
+    along the assigned axis
+
+    Parameters
+    ----------
+    data: 2D array
+
+    axis: int
+        specify which dimension of the data should be normalized
+
+    Returns
+    -------
+    data: 2D array
+        the normalized data
+    """
+    shape = data.shape
+    data = zscore(data, axis=axis, ddof=0)
+    # if zscore fails (standard deviation is zero),
+    # set all values to be zero
+    data = np.nan_to_num(data)
+    data = data / math.sqrt(shape[axis])
+    return data
+
+
+def compute_correlation(set1, set2):
```

Comment:
they are essentially two matrices, depicting two sets of variables. I can use `matrix1/2`